### DISCUSSION TITLE
Why is sketchybar-lua and sketchybar-cpu unused?

### DISCUSSION THREAD

**Discussion by andy-novikov** [shajra/shajra-provisioning#2] [General]
Thanks for sharing your setup. I got here looking for sketchybar-lua example with nix.

I wonder why both cpu and lua extensions do reside in `unused` https://github.com/shajra/shajra-provisioning/blob/f9c2ec5620e52c0d9df12772d3504c06bcb6075a/packages.nix#L310

**Comment by shajra** [comment]
Hi. I am glad to have you browsing around. I'm happy to answer questions.

I normally try to be good about naming, but "unused" is the wrong name. Maybe explaining what's going on can help me come up with a better name.

I like to build all my packages in CI, independent of inclusion in any NixOS-style module. I use `legacyPackages` to expose all my intermediate build stuff in my project's flake. So I can easily access my packages with `nix build .#pkgs.sets…`.

Here are a few ways I then use these packages: 
1. I include a whole package set directly with `home.extraPackages` in a module.
2. The package doesn't override anything in NIxpkgs; I reference it with `pkg.whatever` from within a module.
3. The package does override something in Nixpkgs, and the new package is implicitly picked up by module defaults.

Originally, "unused" was just for case 3.  But that's not the right name.  It's always used, otherwise why build it?  It's more that it's not explicitly referenced in the module.

But then I got case 2 with things like `sketchybar-lua` and `sketchybar-cpu`.   I ended up putting it in this "unused" category, because it was a category of packages that's not "used" by any module's `home.extraPackages`.  But the packages within are used individually.

So I guess "unused" is talking about the category as a whole, not the packages within.  Still, I'm not sure I like that name, because you're right to be confused.

Okay, after all that explanation, I don't yet have a better name.  Maybe one will come to me later.  Or feel free to suggest one.

**Comment by shajra** [comment]
Okay, I came up with a word.  It's "uncategorized."  I committed the change in d61dfb1d.
